src/util/dump.ts

```diff
@@ -90,24 +91,16 @@ export async function downloadAndSaveModule(zimCreator: ZimCreator, mw: MediaWik
     // return :
     //   a promise resolving 1 if data has been succesfully saved or resolving 0 if data was already in Redis
 
-    // the 2 variable functions below are a hack to call startUp() (from module startup) when the 3 generic dependencies (startup, jquery, mediawiki) are loaded.
-    // on wikipedia, startUp() is called in the callback of the call to load.php to dl jquery and mediawiki but since load.php cannot be called in offline,
-    // this hack calls startUp() when custom event fireStartUp is received. Which is dispatched when module mediawiki has finished loading
+    // the function hackStartupModule changes startup script by returning true for all modules so that load.php is not called.
+    // it also removes requestIdleCallback as in our case window is idle after all script tags are called but those script tags
+    // will require the functions which would have been loaded by doPropagation.
     function hackStartUpModule(jsCode: string) {
-        return jsCode.replace(
-            'script=document.createElement(\'script\');',
-            `
-                    document.body.addEventListener('fireStartUp', function () { startUp() }, false);
-                    return;
-                    script=document.createElement('script');`,
-        );
-    }
-    function hackMediaWikiModule(jsCode: string) {
-        jsCode += `(function () {
-            const startUpEvent = new CustomEvent('fireStartUp');
-            document.body.dispatchEvent(startUpEvent);
-        })()`;
-        return jsCode;
+        if (!STARTUP_MODULE_REGEX.test(jsCode)) {
+            throw new Error('unable to hack startup module');
+        }
+
+        return jsCode.replace('mw.requestIdleCallback( doPropagation, { timeout: 1 } );', 'doPropagation();')
+            .replace('function allReady( modules ) {', 'function allReady( modules ) { return true;');
```

Comment:
This is a bit verbose and fragile.  Why we have `if (!STARTUP_MODULE_REGEX.test(jsCode))` in a first place?

Reply:
@kelson42 you asked me to add a check so that if anything in the startup script changes and it stops working we can know about it immediately.

Reply:
OK but this is still super fragile. To test if it "it stops working" you need to secure each the changes your apply really apply.